File: Codility/004_02_FrogRiverOne.cpp
// you can use includes, for example:
// #include <algorithm>
#include <unordered_map>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

int solution(int X, vector<int> &A) {
    // write your code in C++14 (g++ 6.2.0)

    //find the earliest time
    int ans = -1;
    int n = A.size();
    unordered_map<int,bool> mark;

    for(int i = 0; i < n; i++) {
        if(A[i] <= X) {
            mark[A[i]] = true;
            if(mark.size() == X)
                return i;
        }
    }

    return ans;
}

File: Codility/004_02_FrogRiverOne.java
// you can also use imports, for example:
import java.util.*;

// you can write to stdout for debugging purposes, e.g.
// System.out.println("this is a debug message");

class Solution {
    public int solution(int X, int[] A) {
        // write your code in Java SE 8
        HashSet<Integer> frogPath = new HashSet<Integer>();

        int N = A.length;

        for(int i = 0; i < N; i++) {
            if(A[i] <= X) {
                if(frogPath.contains(A[i]) == false)
                    frogPath.add(A[i]);

                if(frogPath.size() == X)
                    return i;
            }
        }

        return -1;
    }
}
